### PR TITLE
Closes #1917 Adding config schema for az_migration module.

### LINKED ISSUE #1917: Add schema for az_migration settings.
## Motivation
Schemas allow you to type data.

### Is your feature request related to a problem? Please describe.
Logic is more confusing without typed data.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Adding a schema file for az_migration module.

## Related issues
Closes #1917 

## How to test
[Enable config inspector](https://db2624a2-98f1-4795-a44e-58012aa2d3f5.probo.build/user?destination=admin/reports/config-inspector)
Search for az_migration
Check that config is correct.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/config/install/schema/az_migration.schema.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      label: 'path to public source files, no trailing or leading slash.'
+    migrate_d7_private_path:
+      type: string
+      label: 'path to private source files, no trailing or leading slash.'
+    migrate_d7_temporary_path:
+      type: string
+      label: 'tmp filepath.'
+    az_migrate_allow_archived_paragraphs:
+      type: boolean
+      label: 'Should we migrate archived paragraphs, not orphans.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/install/schema/az_migration.schema.yml:22]:
```suggestion
      label: 'Should we migrate archived paragraphs, not including orphans.'
```

File: modules/custom/az_migration/config/install/schema/az_migration.schema.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+az_migration.settings:
+  type: config_object
+  label: 'AZ Migration settings'
```
Comment (trackleft) [inline modules/custom/az_migration/config/install/schema/az_migration.schema.yml:3]:
```suggestion
  label: 'AZ Migration Settings'
```

**Review by joeparsons — DISMISSED** [review]

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+az_migration.settings:
+  type: config_object
+  label: 'AZ Migration Settings'
+  mapping:
+    migrate_d7_protocol:
+      type: string
+      label: 'Protocol'
+    migrate_d7_filebasepath:
+      type: string
+      label: 'URL minus schema, no trailing slash.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:10]:
```suggestion
      label: 'URL minus protocol, no trailing slash.'
```

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    migrate_d7_public_path:
+      type: string
+      label: 'path to public source files, no trailing or leading slash.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:13]:
```suggestion
      label: 'Path to public source files, no trailing or leading slash.'
```

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    migrate_d7_private_path:
+      type: string
+      label: 'path to private source files, no trailing or leading slash.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:16]:
```suggestion
      label: 'Path to private source files, no trailing or leading slash.'
```

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    migrate_d7_temporary_path:
+      type: string
+      label: 'tmp filepath.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:19]:
```suggestion
      label: 'Temporary filepath.'
```

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    az_migrate_allow_archived_paragraphs:
+      type: boolean
+      label: 'Should we migrate archived paragraphs, not including orphans.'
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:22]:
```suggestion
      label: 'Migrate archived paragraphs, not including orphans.'
```

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    migrate_d7_public_path:
+      type: string
+      label: 'path to public source files, no trailing or leading slash.'
+    migrate_d7_private_path:
+      type: string
+      label: 'path to private source files, no trailing or leading slash.'
+    migrate_d7_temporary_path:
+      type: string
+      label: 'tmp filepath.'
+    az_migrate_allow_archived_paragraphs:
```
Comment (trackleft) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:20]: Technically this should not exist yet, but if it did, should probably change the name to not have a prefix
```suggestion
    allow_archived_paragraphs:
```

Comment (joeparsons) [reply modules/custom/az_migration/config/schema/az_migration.schema.yml:20]: Was going to suggest this change 👍🏻

File: modules/custom/az_migration/config/schema/az_migration.schema.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+az_migration.settings:
+  type: config_object
+  label: 'AZ Migration Settings'
+  mapping:
+    migrate_d7_protocol:
+      type: string
+      label: 'Protocol'
+    migrate_d7_filebasepath:
+      type: string
+      label: 'Base URL of site where source files live,minus protocol, no trailing slash.'
```
Comment (bberndt-uaz) [inline modules/custom/az_migration/config/schema/az_migration.schema.yml:10]:
```suggestion
      label: 'Base URL of site where source files live, minus protocol, no trailing slash.'
```

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**